platform/darwin/src/NSCoder+MGLAdditions.mm

```diff
@@ -0,0 +1,16 @@
+#import "NSCoder+MGLAdditions.h"
+
+#import "NSArray+MGLAdditions.h"
+
+@implementation NSCoder (MGLAdditions)
+
+- (void)mgl_encodeLocationCoordinates2D:(std::vector<CLLocationCoordinate2D>)coordinates forKey:(NSString *)key {
+    [self encodeObject:[NSArray mgl_coordinatesFromCoordinates:coordinates] forKey:key];
```

Comment:
Now that we have dedicated methods for working with `CLLocationCoordinate2D`s, let’s encode the result of `-[NSValue(MGLAdditions) valueWithMGLCoordinate:]` instead of a dictionary.